bamliquidator_internal/Dockerfile

```diff
@@ -0,0 +1,66 @@
+
+COPY . /opt/liquidator
+
+WORKDIR /opt/liquidator
+RUN make -j6
+
+
+FROM ubuntu:bionic as runner
+
+RUN apt-get -y update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends python \
+    python-numpy python-pip python-tables python-numexpr python-scipy python-setuptools \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN pip install bokeh==0.9.3 "openpyxl>=1.6.1,<2.0.0" \
+    && rm -rf /usr/local/lib/python2.7/dist-packages/bokeh/server/tests
+
+WORKDIR /opt/liquidator
+COPY --from=builder /usr/lib/x86_64-linux-gnu/libtcmalloc_minimal.so.4 \
+                    /usr/lib/x86_64-linux-gnu/libtbb.so.2 \
+                    /usr/lib/x86_64-linux-gnu/libsz.so.2 \
+                    /usr/lib/x86_64-linux-gnu/libhdf5_serial.so.100 \
+                    /usr/lib/x86_64-linux-gnu/libhdf5_serial_hl.so.100 \
+                    /usr/lib/x86_64-linux-gnu/libboost_system.so.1.65.1 \
+                    /usr/lib/x86_64-linux-gnu/libaec.so.0 \
+                    /usr/lib/x86_64-linux-gnu/libhts.so.2 \
+                    /usr/lib/x86_64-linux-gnu/libcurl-gnutls.so.4 \
+                    /usr/lib/x86_64-linux-gnu/libnghttp2.so.14 \
+                    /usr/lib/x86_64-linux-gnu/librtmp.so.1 \
+                    /usr/lib/x86_64-linux-gnu/libpsl.so.5 \
+                    /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2 \
+                    /usr/lib/x86_64-linux-gnu/libkrb5.so.3 \
+                    /usr/lib/x86_64-linux-gnu/libk5crypto.so.3 \
+                    /usr/lib/x86_64-linux-gnu/libkrb5support.so.0 \
+                    /lib/x86_64-linux-gnu/libkeyutils.so.1 \
+                    /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2 \
+                    /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2 \
+                    /usr/lib/x86_64-linux-gnu/libsasl2.so.2 \
+                    /usr/lib/x86_64-linux-gnu/libgssapi.so.3 \
+                    /usr/lib/x86_64-linux-gnu/libheimntlm.so.0 \
+                    /usr/lib/x86_64-linux-gnu/libkrb5.so.26 \
+                    /usr/lib/x86_64-linux-gnu/libasn1.so.8 \
+                    /usr/lib/x86_64-linux-gnu/libhcrypto.so.4 \
+                    /usr/lib/x86_64-linux-gnu/libroken.so.18 \
+                    /usr/lib/x86_64-linux-gnu/libwind.so.0 \
+                    /usr/lib/x86_64-linux-gnu/libheimbase.so.1 \
+                    /usr/lib/x86_64-linux-gnu/libhx509.so.5 \
+                    /lib/x86_64-linux-gnu/
+COPY --from=builder /usr/bin/samtools /usr/bin/
+COPY --from=builder /opt/liquidator/bamliquidator \
+                    /opt/liquidator/bamliquidator_bins \
+                    /opt/liquidator/bamliquidator_regions \
+                    ./
+COPY --from=builder /opt/liquidator/bamliquidatorbatch /opt/liquidator/bamliquidatorbatch
+
+ENV PATH="$PATH:/opt/liquidator"
+
+RUN python bamliquidatorbatch/test.py
+
+ARG GIT_COMMIT
+LABEL git_commit=$GIT_COMMIT
+
+ENTRYPOINT ["bamliquidatorbatch/bamliquidator_batch.py"]
```

Comment:
It's likely going to be common for HPC / researchers using Singularity to want to use this container, and since Singularity doesn't have an understanding of WORKDIR this entrypoint (when converted) will fail because it's a relative path, and not going to be relative to where the user is running the container from. If you change this to be absolute, however, it should convert to Singularity without a hitch!

```
ENTRYPOINT ["/opt/liquidator/bamliquidatorbatch/bamliquidator_batch.py"]
```
or more specifically:

```
ENTRYPOINT ["/usr/bin/python","/opt/liquidator/bamliquidatorbatch/bamliquidator_batch.py"]
```
As the user's python will also likely interfere.

I'm also concerned that you are using python 2 - is there a reason that this is necessary?

And is there a default CMD you would want to use? `--help` or similar?

Reply:
I updated the entrypoint to use absolute paths per your suggestion.

I updated this to use python3.

The default already results in the usage output, which seems ok to me:

```
jdimatteo@x1:~/dev/pipeline/bamliquidator_internal$ docker run bioliquidator/bamliquidator:1.5.0
usage: bamliquidator_batch.py [-h] [-b BIN_SIZE | -r REGIONS_FILE]
                              [-o OUTPUT_DIRECTORY] [-c COUNTS_FILE] [-f]
                              [-e EXTENSION] [--sense {+,-,.}] [-m]
                              [--region_format {gff,bed}] [--skip_plot]
                              [--black_list BLACK_LIST [BLACK_LIST ...]] [-q]
                              [-n NUMBER_OF_THREADS] [--xml_timings]
                              [--version]
                              bam_file_path
bamliquidator_batch.py: error: too few arguments
```

Reply:
Thanks for the suggestion to use singularity -- that looks great. I unfortunately couldn't get it to work in my brief testing, but hopefully the changes to use absolute paths helps others.